File: starter_kits/Scala/hlt/Dropoff.scala
package hlt

object Dropoff {
  def _generate(playerId: PlayerId) = {
    val input = Input.readInput
    val dropoffId = new EntityId(input.getInt)
    val x = input.getInt
    val y = input.getInt
    new Dropoff(playerId, dropoffId, new Position(x, y))
  }
}

class Dropoff(owner: PlayerId, id: EntityId, position: Position) extends Entity(owner, id, position)

File: starter_kits/PHP/hlt/Dropoff.php
<?php

namespace hlt;

class Dropoff extends Entity
{

    public function __construct(PlayerId $owner, EntityId $id, Position $position)
    {
        parent::__construct($owner, $id, $position);
    }

    static function _generate(PlayerId $playerId): Dropoff
    {
        $input = Input::readInput();
        $x = $input->getInt();
        $y = $input->getInt();

        /**
         * @var EntityId
         */
        $dropoffId = new EntityId($input->getInt());

        return new Dropoff($playerId, $dropoffId, new Position($x, $y));
    }
}
